dot/core/service.go

```diff
@@ -523,7 +526,20 @@ func (s *Service) GetMetadata(bhash *common.Hash) ([]byte, error) {
 	}
 
 	rt, err := s.blockState.GetRuntime(bhash)
-	if err != nil {
+	if errors.Is(err, blocktree.ErrFailedToGetRuntime) {
+		rt, err = s.getRuntimeFromDB(*bhash)
+		if err != nil {
+			return nil, fmt.Errorf("getting runtime from database: %w", err)
+		}
+		metadata, err := rt.Metadata()
+		if err != nil {
+			return nil, fmt.Errorf("getting runtime metadata: %w", err)
+		}
+		// ensure the runtime stops and releases resources since it was
+		// instantiated from disk just for this function call.
+		defer rt.Stop()
+		return metadata, nil
```

Comment:
should the `defer rt.Stop()` statement be placed before `metadata, err := rt.Metadata()` to be called in a case where it is not possible to retrieve the `rt.Metadata`